source/common/runtime/runtime_impl.cc

```diff
@@ -552,12 +553,29 @@ void RtdsSubscription::validateUpdateSize(uint32_t num_resources) {
 
 void LoaderImpl::loadNewSnapshot() {
   ThreadLocal::ThreadLocalObjectSharedPtr ptr = createNewSnapshot();
```

Comment:
nit: just use a local var of snapshot shared pointer type, and you can void the dynamic cast below.